### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by guggero**
This PR pull out some commits from https://github.com/lightninglabs/lightning-terminal/pull/848 that don't have any dependency on `lnd v0.18.4` and can be merged immediately.

Didn't add release notes as the improvements are mostly build/development related and don't include any end user features.

### CODE REVIEW

File: perms/mock_dev.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	case "ChainRPC":
+		return &chainrpc.Config{
+			ChainNotifier: &chainreg.NoChainBackend{},
+			Chain:         &mock.ChainIO{},
+		}, true
+	case "DevRPC":
+		return &devrpc.Config{
+			ActiveNetParams: &chaincfg.RegressionNetParams,
+			GraphDB:         &channeldb.ChannelGraph{},
+		}, true
```
Comment (ellemouton) [inline perms/mock_dev.go:68]: 👍

**Review by ellemouton — APPROVED** [review]
lgtm 🙏

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -244,6 +244,19 @@ jobs:
...
+        timeout-minutes: 5
+        run: 7z a logs-itest.zip itest/**/*.log
+
+      - name: Upload log files on failure
+        uses: actions/upload-artifact@v3
+        if: ${{ failure() }}
+        with:
+          name: logs-itest
+          path: logs-itest.zip
+          retention-days: 5
```
Comment (ViktorTigerstrom) [inline .github/workflows/main.yml:258]: Thanks :rocket:!

File: Makefile [hunk 3]
```diff
@@ -94,7 +94,7 @@ DOCKER_TOOLS = docker run \
   -v $(shell bash -c "mkdir -p /tmp/go-lint-cache; echo /tmp/go-lint-cache"):/root/.cache/golangci-lint \
   -v $$(pwd):/build litd-tools
 
-ITEST_TAGS := integration itest $(LND_RELEASE_TAGS)
+ITEST_TAGS := dev integration itest lowscrypt $(LND_RELEASE_TAGS)
```
Comment (ViktorTigerstrom) [inline Makefile:97]: Hmm I'm not fully sure why the `lowscrypt` buildtag is added here? Maybe the commit message could be extended to specify that as well if this is intended :)

File: Makefile [hunk 4]
```diff
@@ -238,6 +238,10 @@ flake-unit:
 	@$(call print, "Flake hunting unit tests.")
 	while [ $$? -eq 0 ]; do GOTRACEBACK=all $(UNIT) -count=1; done
 
+flake-itest-only:
+	@$(call print, "Flake hunting unit tests.")
```
Comment (ViktorTigerstrom) [inline Makefile:242]: Nit:
```suggestion
	@$(call print, "Flake hunting integration tests.")
```

**Review by ViktorTigerstrom — APPROVED** [review]
Thanks a lot for this @guggero :rocket:!

Leaving 1 question and 1 nit 😃

File: Makefile (hunk 3 above)
Comment (guggero) [reply Makefile:97]: Ah, right. Speeds up wallet creation by using weaker encryption. Added that to the commit message.

File: Makefile (hunk 4 above)
Comment (guggero) [reply Makefile]: Thanks, fixed!

**Pull request merged**